flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/calcite/FlinkTypeFactory.scala

```diff
@@ -180,7 +182,37 @@ class FlinkTypeFactory(typeSystem: RelDataTypeSystem)
     * @return a struct type with the input fieldNames, input fieldTypes, and system fields
     */
   def buildLogicalRowType(tableSchema: TableSchema): RelDataType = {
-    buildLogicalRowType(tableSchema.getFieldNames, tableSchema.getFieldTypes)
+    buildLogicalRowType(tableSchema.getFieldNames, tableSchema.getFieldDataTypes)
+  }
+
+  /**
+   * Creates a struct type with the input fieldNames and input fieldTypes using FlinkTypeFactory
+   *
+   * @param fieldNames field names
+   * @param fieldTypes field types, every element is Flink's [[DataType]]
+   * @return a struct type with the input fieldNames, input fieldTypes, and system fields
+   */
+  def buildLogicalRowType(
```

Comment:
revert this?

Reply:
This is needed to fix the view row type inference for legacy planner.